docs/tutorial/flask.rst

```diff
@@ -0,0 +1,530 @@
+
+Now let's keep track of how many visitors your application has received.
+This will require integration with a database to keep the visitor count.
+This will require a few changes:
+
+* We will need to create a database migration that creates the ``visitors`` table
+* We will need to keep track how many times the root endpoint has been called
+  in the database
+* We will need to add a new endpoint to retrieve the number of visitors from the
+  database
+
+Let's start with the database migration to create the required tables.
+The charm created by the ``flask-framework`` extension will execute the
+``migrate.py`` script if it exists. This script should ensure that the
+database is initialised and ready to be used by the application. We will
+create a ``migrate.py`` file containing this logic.
+
+Go back out to the tutorial root directory using ``cd ..``, open the ``migrate.py``
+file using a text editor and paste the following code into it:
+
+.. literalinclude:: code/flask/visitors_migrate.py
+    :language: python
+
+.. note::
+
+    The charm will pass the Database connection string in the
+    ``POSTGRESQL_DB_CONNECT_STRING`` environment variable once
+    postgres has been integrated with the charm.
+
+Open the ``rockcraft.yaml`` file in a text editor and update the version to ``0.3``.
+
+To be able to connect to postgresql from the Flask app the ``psycopg2-binary``
+dependency needs to be added in ``requirements.txt``. The app code also needs
+to be updated to keep track of the number of visitors and to include a new
+endpoint to retrieve the number of visitors to the app. Open ``app.py`` in
+a text editor and replace its contents with the following code:
+
+.. literalinclude:: code/flask/visitors_app.py
+    :language: python
+
+Run ``rockcraft pack`` and upload the newly created rock to the MicroK8s registry:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:docker-2nd-update]
+    :end-before: [docs:docker-2nd-update-end]
+    :dedent: 2
+
+Go back into the charm directory using ``cd charm``. The Flask app now requires
+a database which needs to be declared in the ``charmcraft.yaml`` file. Open
+``charmcraft.yaml`` in a text editor and add the following section to the end:
+
+.. code-block:: yaml
+
+    requires:
+      postgresql:
+        interface: postgresql_client
+        optional: false
+
+Pack the charm using ``charmcraft pack`` and refresh the deployment using Juju:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:refresh-2nd-deployment]
+    :end-before: [docs:refresh-2nd-deployment-end]
+    :dedent: 2
+
+Deploy ``postgresql-k8s`` using Juju and integrate it with ``flask-hello-world``:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:deploy-postgres]
+    :end-before: [docs:deploy-postgres-end]
+    :dedent: 2
+
+Wait for ``juju status`` to show that the App is ``active`` again.
```

Comment:
There's probably too many pieces of inline code for one paragraph. I'd consider splitting it into more digestible chunks.

Reply:
That's a fair point, thanks. I'll make that change